cardano_clusterlib/clusterlib.py

```diff
@@ -1692,6 +1692,42 @@ def get_kes_period(self) -> int:
         """Return last block KES period."""
         return int(self.get_slot_no() // self.slots_per_kes_period)
 
+    def get_kes_period_info(self, opcert_file: FileType) -> dict:
+        """Get information about the current KES period and node's operational certificate.
+
+        Args:
+            issue_op_cert: Operational certificate.
+
+        Returns:
+            dict: A dictionary containing KES period information.
+        """
+        command_output = self.query_cli(["kes-period-info", "--op-cert-file", str(opcert_file)])
+
+        # get output messages
+        messages_str = command_output.split("{")[0]
+        messages_list = []
+
+        if messages_str:
+            message_entry: list = []
+
+            for line in list(filter(None, messages_str.split("\n"))):
```

Comment:
Nice and concise. One comment - I find the `filter` hard to read, especially with `None`. There will be two iterations over the messages lines, one for `filter` and one for your `for` loop. Why not handle it all in your loop then? E.g.

```python
for line in messages_str.split("\n"):
    line = line.strip()  # this will work also for lines that become empty after `strip()`
    if not line:
        continue
    if not message_entry or line[0].isalpha():
        ...
```

Reply:
nice solution :)